### ISSUE TITLE
task statuses don't update after reload

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#155246]
https://github.com/microsoft/vscode/pull/155120#issue-1304051562

**Comment by meganrogge** [comment]
also reported here #155163

**Comment by meganrogge** [comment]
happening bc there's no problem matcher on the configurationProperties

**Comment by meganrogge** [comment]
actually, this is now working - I was just seeing the strange issue in which my VS Code build task doesn't work. @Tyriar could you pls confirm statuses work after reload for the VS  Code build task in today's insider's?



https://user-images.githubusercontent.com/29464607/179244679-62335d24-599d-4a4c-a53a-afb21cec04ed.mov

**Comment by Tyriar** [comment]
@meganrogge I see a spinners when I reload on Windows:

![image](https://user-images.githubusercontent.com/2193314/179250022-0cf58ddf-5366-43ea-9094-26c48f668b0f.png)

Probably also something that needs https://github.com/microsoft/vscode/issues/152634

**Comment by meganrogge** [comment]
yeah @Tyriar if you make a change and save, it'll update the status. before that wasn't happening, which is the bug I'd like help verifying is fixed. works for xterm.js for me

**Comment by meganrogge** [comment]
actually I think it shouldn't be spinning on reload since we will know the status it should have initially. i'll investigate

**Comment by Tyriar** [comment]
This is a little hard to solve actually, note that active problems are also not restored. When reconnecting to the terminal we restore the buffer using `initialText` which doesn't end up firing `onLineData`. We also shouldn't be persisting the status on the pty host because consider the following:

- Fail build, status syncs to pty host
- Restart
- Pass build during restart
- We would restore the fail status here if we persisted to the pty host.

I'm going to move this to August as the fix is a little too involved for me to tackle atm. It's a little annoying but it's also easily worked around (save a change).

I think the fix here should be similar to https://github.com/microsoft/vscode/issues/152634, there should be some mechanism in tasks that allows scanning up to find certain patterns. For https://github.com/microsoft/vscode/issues/152634 we want to scan up a few lines in the buffer and to re-evaluate `endsPattern` if possible, for this issue we want to scan up a reasonable amount on reconnect to find a `beginsPattern` and reparse that and all lines after to activate statuses and problem matchers.
